src/collectors/cephstats/cephstats.py

```diff
@@ -8,12 +8,38 @@
 import re
 import os
 import sys
-
 sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(__file__)),
                                 'ceph'))
 from ceph import CephCollector
 
 
+# This is external to the CephCollector so it can be tested
+# separately.
+def process_ceph_status(output):
+    pattern = re.compile(r'\bclient io .*')
+    res = pattern.search(output)
+    if not res:
+        return {}
+    try:
+        ceph_stats = res.group()
+        number = re.compile(r'\d+')
+        rd = number.search(ceph_stats)
+        if rd:
```

Comment:
That's now how Python works. :)

You're only defining those variables inside this if statement. So if this code branch is never hit because `rd` is falsey, they vars never get defined, resulting in a `NameError` when trying to check them later.